src/main/resources/lib/ssb/dataset/dataset.ts

```diff
@@ -48,11 +48,11 @@ export function refreshDataset(content: Content<DataSource>): CreateOrUpdateStat
   case DataSourceType.STATBANK_API: {
```

Comment:
For clarity sake, we could instead have 2 separate functions instead of lines 45-55 :

```javascript
function extractKey(content: Content<DataSource>) {
  switch(content.dataSource?._selected) {
    case DataSourceType.STATBANK_API: return getStatbankApiKey(content)
    ...
    default: return null
  }
}

function extractData(content: Content<DataSource>) {
  switch(content.dataSource?._selected) {
    case DataSourceType.STATBANK_API: return fetchStatbankApiData(content)
    ...
    default: return null
}

export function refreshDataset(content: Content<DataSource>): CreateOrUpdateStatus {
  let data: JSONstat | TbmlData | null = extractKey(content)
  let key: string | undefined = extractData(content)
  ...
}
```

Reply:
Changed